src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionExtensions.cs

```diff
@@ -22,25 +22,29 @@ public static NuGetVersion[] GetPackageVersions(this IEnumerable<PackageIdentity
                 .ToArray();
         }
 
-        public static IEnumerable<IGrouping<string, NuGetVersion>> GroupById(this IEnumerable<PackageIdentity> packages)
+        public static IEnumerable<IGrouping<string, T>> GroupById<T>(this IEnumerable<T> packages) where T : PackageIdentity
```

Comment:
What is the intention for making this, and the other 3 methods below, "generic", while contraining it to `T : PackageIdentity`?

Is it just to make the line shorter?  I don't think we should be playing "code golf".

Reply:
Sometimes I need that method to work with collections of type `PackageCollectionItem` and other times I need to work with `PackageIdentity` collections